src/app/containers/Swap/saga.ts

```diff
@@ -0,0 +1,139 @@
+    const tokens = yield select(SwapDomains.swapTokens);
+    const fromTokenAddress = tokens[fromTokenSymbol].address;
+    const toTokenAddress = tokens[toTokenSymbol].address;
+    const library = yield select(Web3Domains.selectLibraryDomain);
+    const account = yield select(Web3Domains.selectAccountDomain);
+
+    const swapContract = new Contract(
+      swapRouterAddress,
+      swapRouterABI,
+      getProviderOrSigner(library, account)
+    );
+    const gasEstimate = yield call(
+      swapContract.estimateGas.findBestPathWithGas,
+      amountToGive,
+      fromTokenAddress,
+      toTokenAddress,
+      1,
+      BigNumber.from(225)
+    );
+    const additional = multiply(Number(gasEstimate.toString()), 0.2).toFixed(0);
+    const optimalPath = yield call(
+      swapContract.findBestPathWithGas,
+      amountToGive,
+      fromTokenAddress,
+      toTokenAddress,
+      1,
+      BigNumber.from(225),
+      { gasLimit: (Number(gasEstimate) + Number(additional)).toString() }
+    );
+    yield all([
+      put(SwapActions.setBestPath(optimalPath)),
+      put(SwapActions.setIsGettingBestPath(false)),
+    ]);
+  } catch (error) {
+    console.log(error);
+    yield put(SwapActions.setIsGettingBestPath(false));
+  }
+}
+export function* swap() {
+  try {
+    yield put(SwapActions.setIsSwapping(true));
+    const gasPrice: GenericGasResponse = yield select(GlobalDomains.gasPrice);
+    const library = yield select(Web3Domains.selectLibraryDomain);
+    const account = yield select(Web3Domains.selectAccountDomain);
+    const bestPath: BestPath = yield select(SwapDomains.bestPath);
+    const tokens: ContainerState["tokens"] = yield select(
+      SwapDomains.swapTokens
+    );
+    const tokensList = Object.values(tokens);
+    const swapRouterAddress = yield select(SwapDomains.swapRouterAddress);
+    const swapRouterABI = yield select(SwapDomains.swapRouterABI);
+    const swapRouterContract: SwapRouter = new Contract(
+      swapRouterAddress,
+      swapRouterABI,
+      getProviderOrSigner(library, account)
+    ) as SwapRouter;
+    const infiniteApproval = yield select(SwapDomains.infiniteApproval);
+    const fromTokenAddress = bestPath.path[0];
+    const fromTokenABI = tokensList.find(
+      (token) => token.address === fromTokenAddress
+    )?.ABI;
+    const fromTokenContract = new Contract(
+      fromTokenAddress,
+      fromTokenABI,
+      getProviderOrSigner(library, account)
+    );
+    const amountToReceive = bestPath.amounts[bestPath.amounts.length - 1];
+    const amountToGive = bestPath.amounts[0];
+
+    yield call(
+      checkAndApproveTokenForTrade,
+      fromTokenContract as Erc20,
+      swapRouterAddress,
+      account,
+      amountToGive,
+      infiniteApproval,
+      parseUnits((gasPrice?.gasStandard || "45").toString(), 9), //gasPrice
+      {
+        onTransactionError: () => {
+          toast.error("Transaction failed");
+          throw new Error("Your transaction could not be completed");
```

Comment:
Are we catching this error in the other side? otherwise lets avoid throwing if the caller isnt catching the error.